tools/python/migration-scripts/Ingest_S3_to_postgressql/copy_postgres.py

```diff
@@ -0,0 +1,565 @@
+        logger.info(f"Error retrieving secret {secret_name}: {str(e)}")
+        raise
+    
+    # Parse and return the secret
+    if 'SecretString' in response:
+        secret = response['SecretString']
+        # Parse JSON string into dictionary
+        return json.loads(secret)
+    else:
+        # If binary, decode it (less common)
+        decoded_binary_secret = base64.b64decode(response['SecretBinary'])
+        return json.loads(decoded_binary_secret)
+
+def validate_sql_identifier(identifier):
+    # Simple validation - alphanumeric and underscore only
+    return bool(re.match(r'^[a-zA-Z0-9_]+$', identifier))
+
+
+if __name__ == '__main__':
+
+    # directory = '/data/csv_files/partition*/*.csv'
+    # database_name = 'postgres_testing'
+    # table_name= 'demolarge_restored'
+
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument("-d", "--table", help="postgres table name", required=True)
+    parser.add_argument("-t", "--database", help="postgres database name", required=True)
+    parser.add_argument("-s", "--schema", help="Postgres schema where table resides", default='public', required=False)
+    parser.add_argument("-u", "--user", help="User to login to postgres", default='postgres', required=False)
+    parser.add_argument("-f", "--csv-files-dir", help="CSV files to feed into Postgres", required=True) 
+    parser.add_argument("-e", "--host", help="Postgres Writer Endpoint", required=True)
+    parser.add_argument("-p", "--port", help="Postgres Port", default='5432', required=False)
+    parser.add_argument("-sm", "--secret-name", help="Secrets Manager Secret Name", required=True)
+    parser.add_argument("-pt", "--parallel-threads", help = "Number of threads that will ingest CSV files parallely", default=10, required = False)
+    parser.add_argument("-pd", "--processed-dir", help = "location for moving the processed files",default = None, required = False)
+    parser.add_argument("-ld", "--logs_dir", help='Directory for postgres ingestion logs (default: postgres-ingestion-logs)', default = None, required = False)
+    parser.add_argument("-sns","--sns_topic-arn", help="SNS topic ARN for sending any batchload failures", default=None, required=False)
+
+
+    args = parser.parse_args()
+
+    log_dir = args.logs_dir
+
+    if log_dir is None:
+        log_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'postgres-ingestion-logs')
+    os.makedirs(log_dir, exist_ok=True)
+    custom_logger_file = f"postgres_ingestion_{datetime.now().strftime('%Y%m%d_%H%M%S')}.log"
+    logger = create_logger('postgres_ingestion', log_file=f"{log_dir}/{custom_logger_file}" )
+    logger.info(f"logging into {log_dir}/{custom_logger_file}")
+
+    #complete parsing
+    table_name = args.table
+    database_name = args.database
+    directory = args.csv_files_dir
+    schema = args.schema
+    host = args.host
+    port = args.port
+    user = args.user
+    num_of_threads = args.parallel_threads
```

Comment:
need to cast as int (currently parsed as a string)
```suggestion
    num_of_threads = int(args.parallel_threads)
```